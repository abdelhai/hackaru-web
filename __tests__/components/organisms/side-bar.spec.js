import Factory from '@/__tests__/__setups__/factory';
import SideBar from '@/components/organisms/side-bar';

describe('SideBar', () => {
  let factory;
  let wrapper;

  beforeEach(() => {
    factory = new Factory(SideBar);
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  describe('when click add button', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find('.add-button').trigger('click');
    });

    it('show new activity modal', () => {
      expect(factory.options.mocks.$modal.show).toHaveBeenCalledWith(
        'activity'
      );
    });
  });
});
