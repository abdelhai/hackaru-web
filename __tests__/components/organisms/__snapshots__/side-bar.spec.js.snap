// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar render correctly 1`] = `
<section
  class="menu"
>
  <h1>
    <a
      aria-label="ariaLabels.index"
    >
      <img
        class="logo-icon"
        src="@/assets/logo.svg"
      />
    </a>
  </h1>
   
  <span
    class="url"
  >
    www.hackaru.app
  </span>
   
  <ul>
    <li
      class=""
    >
      <a
        aria-label="ariaLabels.index"
      >
        <icon-stub
          class="icon is-small"
          name="home-icon"
        />
      </a>
    </li>
    <li
      class=""
    >
      <a
        aria-label="ariaLabels.calendar"
      >
        <icon-stub
          class="icon is-small"
          name="calendar-icon"
        />
      </a>
    </li>
    <li
      class=""
    >
      <a
        aria-label="ariaLabels.reports"
      >
        <icon-stub
          class="icon is-small"
          name="bar-chart-icon"
        />
      </a>
    </li>
    <li
      class=""
    >
      <a
        aria-label="ariaLabels.settings"
      >
        <icon-stub
          class="icon is-small"
          name="settings-icon"
        />
      </a>
    </li>
  </ul>
   
  <button
    aria-label="ariaLabels.addActivity"
    class="add-button"
    type="button"
  >
    <icon-stub
      class="is-primary"
      name="clock-icon"
    />
  </button>
</section>
`;
